subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskPropertyUtils.java

```diff
@@ -21,13 +21,15 @@
 import org.gradle.api.internal.tasks.properties.PropertyVisitor;
 import org.gradle.api.internal.tasks.properties.PropertyWalker;
 import org.gradle.internal.reflect.TypeValidationContext;
+import org.gradle.internal.scan.UsedByScanPlugin;
 
 @NonNullApi
 public class TaskPropertyUtils {
     /**
      * Visits both properties declared via annotations on the properties of the task type as well as
      * properties declared via the runtime API ({@link org.gradle.api.tasks.TaskInputs} etc.).
      */
+    @UsedByScanPlugin("test-distribution")
```

Comment:
We've talked briefly about this with @britter, @wolfs and @jonnybbb. I think it would be nice to find a better way to declare this dependency, i.e. by introducing a new subproject that we keep backwards compatible, like we do with the `:enterprise` subproject.

Not something we need to fix in this PR, but it's a good example to start the discussions from.